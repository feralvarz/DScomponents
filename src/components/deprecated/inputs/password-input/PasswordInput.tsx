import React, { forwardRef, useState } from 'react';
import { Password } from 'primereact/password';
import type { PasswordType } from '../../../../types/components/password/PasswordInputType';

const getSizeClasses = (size: any) => {
  switch (size) {
    case 'small': {
      return 'p-inputtext-sm';
    }
    case 'large': {
      return 'p-inputtext-lg';
    }
    case 'medium': {
      return undefined;
    }
    default: {
      return undefined;
    }
  }
};

/**
 * @deprecated use Input[type="password"] from components/Input
 */
function PasswordInput(props: PasswordType, ref: any) {
  const {
    id,
    label,
    theme = 'main',
    required,
    disabled = false,
    showFeedBack = false,
    toggleMask = true,
    placeholder,
    className,
  } = props;
  const [value, setValue] = useState<string>('');
  const size = getSizeClasses(props.size);
  const validClass = props.valid ? '' : 'p-invalid';
  const passClassName = `${size} ${validClass} ${className} `;
  return (
    <div className="form-control w-full bg-transparent" data-theme={theme}>
      <label className={`${disabled ? 'text-neutral-400' : ''} label`} htmlFor={id}>
        <p>
          {label}{' '}
          <span className={`${disabled ? 'text-neutral-400' : 'text-secondary'}`}>
            {required ? '*' : null}
          </span>
        </p>
      </label>
      <div>
        <Password
          aria-describedby={`${id}-help`}
          className={passClassName}
          disabled={disabled}
          feedback={showFeedBack}
          placeholder={placeholder}
          ref={ref}
          toggleMask={toggleMask}
          value={value}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) => setValue(e.target.value)}
          {...props}
        />
      </div>
    </div>
  );
}

export default forwardRef<Password, PasswordType>(PasswordInput);
