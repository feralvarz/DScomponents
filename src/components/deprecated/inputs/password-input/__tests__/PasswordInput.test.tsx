import { render, screen } from '@testing-library/react';
import PasswordInput from '../PasswordInput';
import '@testing-library/jest-dom';

describe('PasswordInput', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<PasswordInput id="password" theme="main" />);
    expect(container).toMatchSnapshot();
  });

  it('renders label', () => {
    render(<PasswordInput id="password" label="Password" theme="main" required />);
    expect(screen.getByText('Password')).toBeInTheDocument();
  });

  it('should render the placeholder', () => {
    const { getByPlaceholderText } = render(
      <PasswordInput
        id="input"
        label="Password Test Input"
        placeholder="Placeholder"
        theme="main"
      />,
    );
    expect(getByPlaceholderText('Placeholder')).toBeInTheDocument();
  });
});
