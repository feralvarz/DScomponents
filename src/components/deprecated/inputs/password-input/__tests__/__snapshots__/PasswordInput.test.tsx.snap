// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput renders correctly with default props 1`] = `
<div>
  <div
    class="form-control w-full bg-transparent"
    data-theme="main"
  >
    <label
      class=" label"
      for="password"
    >
      <p>
         
        <span
          class="text-secondary"
        />
      </p>
    </label>
    <div>
      <div
        class="p-password p-component p-inputwrapper p-input-icon-right undefined p-invalid undefined "
        id="password"
      >
        <input
          aria-describedby="password-help"
          class="p-inputtext p-component p-password-input"
          theme="main"
          type="password"
          value=""
        />
        <i
          class="pi pi-eye"
        />
      </div>
    </div>
  </div>
</div>
`;
